File: src/main/java/com/gina/simulator/simulation/Simulation.java
package com.gina.simulator.simulation;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.gina.simulator.enums.Difficulty;
import com.gina.simulator.incident.Incident;
import com.gina.simulator.person.Person;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
public class Simulation {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @JsonIgnoreProperties({"password"}) // doesnt work ??
    @ManyToOne
    private Person person;

    @JsonIgnore
    @OneToMany(mappedBy = "simulation")
    private List<Incident> incidents;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private int rating;

    private Difficulty difficulty;
}

File: src/main/client/src/models/Simulation.ts
export interface ISimulation {
    id: string;
    person?: {
        username: string;
    };
    startTime: Date;
    endTime?: Date;
    rating: number;
    difficulty: string;
}

export class Simulation implements ISimulation {
    public id: string;
    public startTime: Date;
    public endTime?: Date;
    public rating: number;
    public difficulty: string;

    constructor(data: ISimulation) {
        this.id = data.id;
        this.startTime = new Date(data.startTime);
        if (data.endTime) {
            this.endTime = new Date(data.endTime);
        }
        this.rating = data.rating;
        this.difficulty = data.difficulty;
    }

    public isActive(): boolean {
        return !this.endTime;
    }
}